api/v1alpha1/extension_types.go

```diff
@@ -0,0 +1,133 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ExtensionManagedState string
+
+const (
+	// Pause resolution of this Extension
+	ManagedStateActive ExtensionManagedState = "Active"
+	// Peform resolution of this Extension
+	ManagedStatePaused ExtensionManagedState = "Paused"
+)
+
+type ExtensionSourcePackage struct {
+	//+kubebuilder:validation:MaxLength:=48
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+(-[a-z0-9]+)*$
+	Name string `json:"name"`
+
+	//+kubebuilder:validation:MaxLength:=64
+	//+kubebuilder:validation:Pattern=`^(\s*(=||!=|>|<|>=|=>|<=|=<|~|~>|\^)\s*(v?(0|[1-9]\d*|[x|X|\*])(\.(0|[1-9]\d*|x|X|\*]))?(\.(0|[1-9]\d*|x|X|\*))?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?)\s*)((?:\s+|,\s*|\s*\|\|\s*)(=||!=|>|<|>=|=>|<=|=<|~|~>|\^)\s*(v?(0|[1-9]\d*|x|X|\*])(\.(0|[1-9]\d*|x|X|\*))?(\.(0|[1-9]\d*|x|X|\*]))?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?)\s*)*$`
+	//+kubebuilder:Optional
+	// Version is an optional semver constraint on the package version. If not specified, the latest version available of the package will be installed.
+	// If specified, the specific version of the package will be installed so long as it is available in any of the content sources available.
+	// Examples: 1.2.3, 1.0.0-alpha, 1.0.0-rc.1
+	//
+	// For more information on semver, please see https://semver.org/
+	Version string `json:"version,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=48
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+([\.-][a-z0-9]+)*$
+	// Channel constraint definition
+	Channel string `json:"channel,omitempty"`
+}
+
+// TODO: Implement ExtensionSourceDirect containing a URL or other reference mechanism
+
+type ExtensionSource struct {
+	// A source package defined by a name, version and/or channel
+	Package *ExtensionSourcePackage `json:"package,omitempty"`
+
+	// A direct package defined by a URL
+	// TODO: Direct *ExtensionSourceDirect `json:"direct,omitempty"`
+}
+
+// ExtensionSpec defines the desired state of Extension
+type ExtensionSpec struct {
+	//+kubebuilder:validation:Enum:=Active;Paused
+	//+kubebuilder:default:=Active
+	//+kubebuilder:Optional
+	//
+	// Pause reconciliation on this Extension
+	Managed ExtensionManagedState `json:"managed,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=64
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+([\.-][a-z0-9]+)*$
+	//
+	// ServiceAccount name used to install this extension
+	ServiceAccountName string `json:"serviceAccountName"`
+
+	//+kubebuilder:validation:MaxLength:=64
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+([\.-][a-z0-9]+)*$
+	//+kubebuilder:Optional
+	//
+	// Location of installation TBD??
+	DefaultNamespace string `json:"defaultNamespace,omitempty"`
```

Comment:
Remove for now?

Reply:
The discussion above (@kevinrizza) was to keep it

Reply:
If we are planning to keep it, can we elaborate it further by mentioning that it is the default namespace where Extension resources are installed, and it defaults to the ns where the Extension is present.

Also do we need the max length validation for namespace? I think the allowed resource name limit of [253](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names) characters applies to ns too.